src/Ztm.WebApi/ICallbackRepository.cs

```diff
@@ -0,0 +1,15 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Net;
+
+namespace Ztm.WebApi
+{
+    public interface ICallbackRepository
+    {
+        Task<Callback> AddAsync(IPAddress ip, DateTime registeredTime, Uri url, CancellationToken cancellationToken);
+        Task<Callback> SetStatusAsCompletedAsync(Guid id, CancellationToken cancellationToken);
```

Comment:
IMO just `SetCompletedAsyc` should be enough.

Reply:
make sense